infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftInternalRuntime.java

```diff
@@ -139,6 +128,54 @@ protected void internalStart(Map<String, String> startOptions) throws Infrastruc
     }
   }
 
+  /**
+   * Bootstraps machines in parallel.
+   *
+   * @throws InfrastructureException when any error occurs while bootstrapping machines
+   * @throws InterruptedException when machines bootstrapping was interrupted
+   */
+  private void bootstrapMachines() throws InfrastructureException, InterruptedException {
+    CompletableFuture<InfrastructureException> firstFailed = new CompletableFuture<>();
+    List<CompletableFuture> checkMachines = new ArrayList<>(machines.size());
+    for (OpenShiftMachine machine : machines.values()) {
+      checkMachines.add(
+          CompletableFuture.runAsync(
+                  () -> {
+                    try {
+                      machine.waitRunning(machineStartTimeoutMin);
```

Comment:
It's needed to rethrow `InterruptedException` while waiting of machine instead of wrapping it into `InfrastructureException`.

Reply:
Can you show where is wrapped? Test shows it rethrown correctly;

Reply:
Here https://github.com/eclipse/che/blob/f9381a6533fba905b73e311802a4a1802761bfd0/infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftPods.java#L184-L186

Reply:
How it's related to my PR ?

Reply:
If `Thread` will be interrupted while waiting running you will rethrow the wrong exception.

Reply:
That is the problem of `wait()` method i think.

Reply:
I think so. And you can fix it in your PR, it's so easy I think. Or create a new issue for that.

Reply:
Uff. You're confused me strongly. I thought you men that  _MY_ code changes the ex.

Reply:
Are you going to throw `InterruptedException` from `OpenShiftPods#wait` method?

Reply:
I don't know. I'm not familiar with that code.